### ISSUE TITLE
Use port 8883 without SSL

### ISSUE THREAD

**Issue by llamafilm** [eclipse/mosquitto#2180]
While trying to debug some complex network issues, I thought it would be helpful to run mosquitto on port 8883 without using SSL.  Unfortunately this doesn't work.  Any other port number works fine, but it seems like when you specify 8883 it automatically uses SSL and I can't find a way to disable it.
I'm using version 2.0.10 installed from homebrew.  Is this a bug or a feature request?

mosquitto.conf contents:

```
allow_zero_length_clientid true
listener 8883
protocol mqtt
allow_anonymous true
```

Server output:
```
% /usr/local/sbin/mosquitto -c /usr/local/Cellar/mosquitto/2.0.10/etc/mosquitto/mosquitto.conf -v
1618468467: mosquitto version 2.0.10 starting
1618468467: Config loaded from /usr/local/Cellar/mosquitto/2.0.10/etc/mosquitto/mosquitto.conf.
1618468467: Opening ipv6 listen socket on port 8883.
1618468467: Opening ipv4 listen socket on port 8883.
1618468467: mosquitto version 2.0.10 running
1618468474: New connection from 192.168.4.242:51927 on port 8883.
1618468474: Client <unknown> disconnected due to protocol error.
```

Client output:
```
                                                                                                                                    elliott@fifteen liftmaster % mosquitto_sub -L mqtt://192.168.4.242:8883/topic --debug           
Client (null) sending CONNECT
Error: MOSQ_OPT_SSL_CTX_WITH_DEFAULTS used without specifying cafile, capath or psk.
```

**Comment by ralight** [comment]
Does that work for you? :)

**Comment by llamafilm** [comment]
Wow that was fast, thanks!  Sound like it should work, but I'm having trouble building this on macOS.  I checked out the develop branch and ran cmake with no issue.  The build shows a few errors:

- 'openssl/opensslconf.h' file not found
- use of undeclared identifier 'TCP_KEEPIDLE'
- ld: cannot link directly with dylib/framework, your binary is not an allowed client of /usr/lib/libcrypto.dylib for architecture x86_64

Here's the build output.  The missing openssl is probably a mistake on my part, but I'm not sure about the other two.

```
% make -j8 -I/usr/local/Cellar/openssl@1.1/1.1.1k/include         
[  0%] Building C object plugins/examples/message-timestamp/CMakeFiles/mosquitto_message_timestamp.dir/mosquitto_message_timestamp.c.o
[  2%] Building C object plugins/examples/add-properties/CMakeFiles/mosquitto_add_properties.dir/mosquitto_add_properties.c.o
[  2%] Built target mosquitto_client_properties
[  3%] Built target mosquitto_auth_by_ip
[  9%] Built target mosquitto_dynamic_security
[ 10%] Linking C executable mosquitto_passwd
Consolidate compiler generated dependencies of target mosquitto_connection_state
Consolidate compiler generated dependencies of target mosquitto_payload_modification
In file included from /Users/elliott/source/mosquitto/plugins/examples/add-properties/mosquitto_add_properties.c:35:
/Users/elliott/source/mosquitto/config.h:52:12: fatal error: 'openssl/opensslconf.h' file not found
#  include <openssl/opensslconf.h>
           ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/elliott/source/mosquitto/plugins/examples/message-timestamp/mosquitto_message_timestamp.c:31:
/Users/elliott/source/mosquitto/config.h:52:12: fatal error: 'openssl/opensslconf.h' file not found
#  include <openssl/opensslconf.h>
           ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[ 11%] Built target mosquitto_payload_modification
[ 12%] Built target mosquitto_connection_state
make[2]: *** [plugins/examples/message-timestamp/CMakeFiles/mosquitto_message_timestamp.dir/mosquitto_message_timestamp.c.o] Error 1
make[1]: *** [plugins/examples/message-timestamp/CMakeFiles/mosquitto_message_timestamp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
1 error generated.
make[2]: *** [plugins/examples/add-properties/CMakeFiles/mosquitto_add_properties.dir/mosquitto_add_properties.c.o] Error 1
make[1]: *** [plugins/examples/add-properties/CMakeFiles/mosquitto_add_properties.dir/all] Error 2
[ 12%] Generating mosquitto-tls.7
Consolidate compiler generated dependencies of target mosquitto
ld: cannot link directly with dylib/framework, your binary is not an allowed client of /usr/lib/libcrypto.dylib for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [apps/mosquitto_passwd/mosquitto_passwd] Error 1
make[1]: *** [apps/mosquitto_passwd/CMakeFiles/mosquitto_passwd.dir/all] Error 2
[ 13%] Generating mosquitto_rr.1
[ 14%] Generating mosquitto.8
[ 15%] Linking C shared library libmosquitto.dylib
ld: cannot link directly with dylib/framework, your binary is not an allowed client of /usr/lib/libcrypto.dylib for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libmosquitto.2.0.8.dylib] Error 1
make[1]: *** [lib/CMakeFiles/libmosquitto.dir/all] Error 2
[ 16%] Building C object src/CMakeFiles/mosquitto.dir/bridge.c.o
[ 16%] Building C object src/CMakeFiles/mosquitto.dir/__/lib/handle_pubrec.c.o
[ 16%] Building C object src/CMakeFiles/mosquitto.dir/__/lib/handle_pubrel.c.o
[ 17%] Building C object src/CMakeFiles/mosquitto.dir/__/lib/handle_suback.c.o
[ 17%] Building C object src/CMakeFiles/mosquitto.dir/handle_subscribe.c.o
/Users/elliott/source/mosquitto/src/bridge.c:184:42: error: use of undeclared identifier 'TCP_KEEPIDLE'
                setsockopt(context->sock, IPPROTO_TCP, TCP_KEEPIDLE, (const void*)&idle, sizeof(idle)) ||
                                                       ^
[ 18%] Building C object src/CMakeFiles/mosquitto.dir/__/lib/handle_unsuback.c.o
1 error generated.
[ 18%] Building C object src/CMakeFiles/mosquitto.dir/handle_unsubscribe.c.o
[ 19%] Building C object src/CMakeFiles/mosquitto.dir/keepalive.c.o
[ 19%] Building C object src/CMakeFiles/mosquitto.dir/logging.c.o
make[2]: *** [src/CMakeFiles/mosquitto.dir/bridge.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/CMakeFiles/mosquitto.dir/all] Error 2
```

**Comment by llamafilm** [comment]
Okay I figured it out.  I had to edit CMakeLists.txt to hardcode the path to openssl, thanks to #1923.
Then I deleted the line about TCP_KEEPIDLE in bridge.c and ran `cmake -D DOCUMENTATATION=no` because it was showing a bunch of xsl errors, even though I have docbook-xsl and libxslt from homebrew.

The TCP_KEEPIDLE thing will be a problem on any Mac, but I'll ignore that for now since it's in dev branch.
